index.bs

```diff
@@ -1019,6 +1030,10 @@ spec:css-syntax-3;
             {{CredentialMediationRequirement/conditional}} and a `null` credential is discovered,
             promise |p| is not resolved.
 
+    1. [=map/For each=] |optionKey| → <var ignore>optionValue</var> of |options|:
+
+        1. [=set/Remove=] |optionKey| from |settings|' [=active credential types=].
```

Comment:
Clearing the `active credential types` set should happen after the promise is resolved or rejected, not before the promise is returned. The promise is returned immediately, so from the perspective of the website `active credential types` would always be empty, even with a pending request (and therefore, a pending promise).

I think what you want is to [react](https://webidl.spec.whatwg.org/#dfn-perform-steps-once-promise-is-settled) to the promise instead.

Reply:
Done